modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpIgniteClient.java

```diff
@@ -125,6 +126,11 @@ private TcpIgniteClient(ClientConfiguration cfg) throws ClientException {
             ch.channelsInit();
 
             ch.addChannelFailListener(metadataHnd::onReconnect);
+            // Binary descriptors and user types caches must be cleared so that the
+            // client will register all the user types within the cluster once again in case this information
+            // was lost during the cluster failover.
+            ch.addChannelFailListener(marshCtx::clearUserTypesCache);
```

Comment:
We can add use one fail listener to perform all tasks instead of adding three of them.

Reply:
Done